src/Workspaces/Core/Portable/Options/IOption.cs

```diff
@@ -11,5 +12,7 @@ public interface IOption
         Type Type { get; }
         object DefaultValue { get; }
         bool IsPerLanguage { get; }
+
+        ImmutableArray<OptionSerialization> Serializations { get; }
```

Comment:
It is supposed to be "dataish". Right now it's pretty much data, but for editor config options will probably have small functions for converting from strings to the underlying enum types. So small amounts of code are fine, but those shouldn't have any interesting dependencies and should be _very_ cheap.

Doing "all options" might be terribly expensive, since that'll potentially have to do lots of editorconfig walking and other stuff we don't want to happen. I suspect some sort of lazy approach is the right thing to do.